### PR TITLE
test/system/250-systemd.bats: fix flake

### PR DESCRIPTION
**Pull request by vrothberg**
Fix a flake in the kube-template test.  After stopping the service, we want to make sure that the service container gets removed.  However, ther is a small race window. `systemctl stop` will return when the service container _exits_.  In between that and the `container exists` check, the service container may have not yet been removed.  Hence, add a loop to account for that race.

Fixes: #16047
Signed-off-by: Valentin Rothberg <vrothberg@redhat.com>

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by vrothberg** [comment]
@containers/podman-maintainers PTAL

**Review by giuseppe — APPROVED** [review]
LGTM

**Review by Luap99 — COMMENTED** [review]
I always thought `systemctl stop` and `systemctl start` are blocking? So I assume the commands in the unit should have been all be finished by the time the systemctl command finishes.

**Comment by vrothberg** [comment]
> I always thought `systemctl stop` and `systemctl start` are blocking? So I assume the commands in the unit should have been all be finished by the time the systemctl command finishes.

You made me doubt.  I will do some more/better research with more coffee tomorrow..

/hold

**Comment by Luap99** [comment]
I found this in `systemctl(1)`:
> --no-block
           Do not synchronously wait for the requested operation to finish. If this is not specified, the job will be verified, enqueued and systemctl will wait until the unit's start-up is completed. By passing this
           argument, it is only verified and enqueued. This option may not be combined with --wait.

It only mentions the start part but not stop.

File: test/system/250-systemd.bats
Comment (rhatdan) [inline test/system/250-systemd.bats:446]: Could you do a `podman wait $service_container`

**Comment by vrothberg** [comment]
Closing.  The test _is_ doing the right thing.  `systemctl stop` is executing `kube down`, so the service container should have been removed already.

Must be something else but I was chasing a ghost.

**Pull request closed without merging**
